### PR TITLE
Use new entities field in FieldMetadata

### PR DESCRIPTION
**Pull request by carlesonielfa**
### Description
* Adds ingestion of entities in FieldMetadata through the new field with fallback to the legacy field
* Sets the stage to deprecate the use of the fields `ner` and `positions`
* Keeps resource API backwards compatibility

The changes are compatible with
- current state (processor entities in legacy field + data augmentation entities in new)
- processor entities in both new and legacy fields 
- using only the new fields

### How was this PR tested?
Modified existing tests and fixtures to use the new field. Extended ingestion tests to test that data augmentation entities were being ingested

### CODE REVIEW

File: nucliadb/src/nucliadb/ingest/orm/brain.py [hunk 1]
```diff
@@ -498,24 +498,54 @@ def process_field_metadata(
...
+            for ent in entities.entities:
+                entity_text = ent.text
+                entity_label = ent.label
+                # Seems like we don't care about where the entity is in the text
+                # entity_positions = entity.positions
+                labels["e"].add(
+                    f"{entity_label}/{entity_text}"
+                )  # Add data_augmentation_task_id as a prefix?
+                relation_node_entity = RelationNode(
+                    value=entity_text,
```
Comment (jotare) [inline nucliadb/src/nucliadb/ingest/orm/brain.py:518]: Is this change on purpose? In the legacy version, `value` was the `label/text`, while now is only `text`.

File: nucliadb/src/nucliadb/ingest/orm/resource.py [hunk 2]
```diff
@@ -1254,3 +1256,25 @@ def extract_field_metadata_languages(
...
+    target_entites_dict: Union[dict[str, str], ScalarMap[str, str]], field_metadata: FieldMetadata
+):
+    """
+    Update the entities dict with the entities from the field metadata.
+    Method created to ease the transition from legacy ner field to new entities field.
+    """
+    # Data Augmentation + Processor entities
+    # This will overwrite entities detected from more than one data augmentation task
+    # TODO: Change TrainMetadata proto to accept multiple entities with the same text
+    ents = {
```
Comment (jotare) [inline nucliadb/src/nucliadb/ingest/orm/resource.py:1271]: nit: prefer the full name (like `entities`) rather than a partial word like `ents`

File: nucliadb/src/nucliadb/ingest/orm/brain.py (hunk 1 above)
Comment (carlesonielfa) [reply nucliadb/src/nucliadb/ingest/orm/brain.py:518]: Maybe I am mistaken, but in the legacy version I see that the value=`entity` and `entity` is only the text
```python
entity_array = klass_entity.split("/")
klass = entity_array[0] # This is the label
entity = "/".join(entity_array[1:]) # This takes everything except the label (so the text)
```

File: nucliadb/src/nucliadb/ingest/orm/resource.py [hunk 3]
```diff
@@ -1254,3 +1256,25 @@ def extract_field_metadata_languages(
     for _, splitted_metadata in field_metadata.metadata.split_metadata.items():
         languages.add(splitted_metadata.language)
     return list(languages)
+
+
+def _update_entities_dict(
+    target_entites_dict: Union[dict[str, str], ScalarMap[str, str]], field_metadata: FieldMetadata
```
[the remark this answers is not in the document]
Comment (carlesonielfa) [reply nucliadb/src/nucliadb/ingest/orm/resource.py]: fixed! Good catch! I did not know the protobuf type implemented Mapping

File: nucliadb/src/nucliadb/ingest/orm/resource.py (hunk 2 above)
Comment (carlesonielfa) [reply nucliadb/src/nucliadb/ingest/orm/resource.py]: Fixed!

**Review by jotare — APPROVED** [review]
